### PR TITLE
Renaming to QueueProperties, creating CreateQueueOptions, flattening MessageDetailsCount

### PR DESCRIPTION
**Pull request by conniey**
* Renaming to QueueProperties
* Creating CreateQueueOptions
* Flattening MessageDetailsCount
* Rename to TotalMessageCount

Related to #13000

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/CreateQueueBodyContent.java [hunk 1]
```diff
@@ -52,18 +52,18 @@ public CreateQueueBodyContent setType(String type) {
      *
      * @return the queueDescription value.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/CreateQueueBodyContent.java:53]: Change this to `queueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/CreateQueueBodyContent.java [hunk 2]
```diff
@@ -52,18 +52,18 @@ public CreateQueueBodyContent setType(String type) {
...
      */
-    public QueueDescription getQueueDescription() {
-        return this.queueDescription;
+    public QueueProperties getQueueProperties() {
+        return this.queueProperties;
     }
 
     /**
      * Set the queueDescription property: Properties of the new queue.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/CreateQueueBodyContent.java:60]: Change this to `queueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionEntryContent.java [hunk 3]
```diff
@@ -52,18 +52,18 @@ public QueueDescriptionEntryContent setType(String type) {
      *
      * @return the queueDescription value.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionEntryContent.java:53]: Change this to `queueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionEntryContent.java [hunk 4]
```diff
@@ -52,18 +52,18 @@ public QueueDescriptionEntryContent setType(String type) {
...
-        return this.queueDescription;
+    public QueueProperties getQueueProperties() {
+        return this.queueProperties;
     }
 
     /**
      * Set the queueDescription property: Description of a Service Bus queue resource.
      *
-     * @param queueDescription the queueDescription value to set.
+     * @param queueProperties the queueDescription value to set.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionEntryContent.java:62]: Change this to `queueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionEntryContent.java [hunk 5]
```diff
@@ -52,18 +52,18 @@ public QueueDescriptionEntryContent setType(String type) {
...
      * @return the queueDescription value.
      */
-    public QueueDescription getQueueDescription() {
-        return this.queueDescription;
+    public QueueProperties getQueueProperties() {
+        return this.queueProperties;
     }
 
     /**
      * Set the queueDescription property: Description of a Service Bus queue resource.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionEntryContent.java:60]: Change this to `queueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionResponseContent.java [hunk 6]
```diff
@@ -52,18 +52,18 @@ public QueueDescriptionResponseContent setType(String type) {
      *
      * @return the queueDescription value.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionResponseContent.java:53]: Change this to `queueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionResponseContent.java [hunk 7]
```diff
@@ -52,18 +52,18 @@ public QueueDescriptionResponseContent setType(String type) {
...
      */
-    public QueueDescription getQueueDescription() {
-        return this.queueDescription;
+    public QueueProperties getQueueProperties() {
+        return this.queueProperties;
     }
 
     /**
      * Set the queueDescription property: Description of a Service Bus queue resource.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionResponseContent.java:60]: Change this to `queueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java [hunk 8]
```diff
@@ -0,0 +1,497 @@
...
+        return name;
+    }
+
+    /**
+     * Get the autoDeleteOnIdle property: ISO 8601 timeSpan idle interval after which the queue is automatically
+     * deleted. The minimum duration is 5 minutes.
+     *
+     * @return the autoDeleteOnIdle value.
+     */
+    public Duration getAutoDeleteOnIdle() {
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java:162]: Should this be named `getAutoDeleteDurationOnIdle()`?

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java [hunk 9]
```diff
@@ -0,0 +1,497 @@
...
+        return this;
+    }
+
+    /**
+     * Get the deadLetteringOnMessageExpiration property: A value that indicates whether this queue has dead letter
+     * support when a message expires.
+     *
+     * @return the deadLetteringOnMessageExpiration value.
+     */
+    public boolean deadLetteringOnMessageExpiration() {
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java:210]: Missing `is` prefix.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java [hunk 10]
```diff
@@ -0,0 +1,497 @@
...
+        return this;
+    }
+
+    /**
+     * Get the enablePartitioning property: A value that indicates whether the queue is to be partitioned across
+     * multiple message brokers.
+     *
+     * @return the enablePartitioning value.
+     */
+    public boolean enablePartitioning() {
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java:279]: `isPartitioningEnabled`?

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/QueueProperties.java [hunk 11]
```diff
@@ -15,16 +15,14 @@
 import java.time.OffsetDateTime;
 import java.util.ArrayList;
 import java.util.List;
-
-import static com.azure.messaging.servicebus.implementation.ServiceBusConstants.DEFAULT_LOCK_DURATION;
-import static com.azure.messaging.servicebus.implementation.ServiceBusConstants.MAX_DURATION;
+import java.util.Objects;
 
 /** The QueueDescription model. */
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/QueueProperties.java:20]: Change this to `QueueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/QueueProperties.java [hunk 12]
```diff
@@ -337,7 +326,7 @@ public String getName() {
      * @param queueName Name of the queue.
      * @return the QueueDescription object itself.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/QueueProperties.java:327]: Update all javadocs in this class to `queueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/QueueRuntimeInfo.java [hunk 13]
```diff
@@ -14,29 +15,39 @@
...
+    private final int transferDeadLetterMessageCount;
+    private final int transferMessageCount;
 
     /**
      * Creates a new instance with runtime properties extracted from the given QueueDescription.
      *
-     * @param queueDescription Queue description to extract runtime information from.
+     * @param queueProperties Queue description to extract runtime information from.
      *
      * @throws NullPointerException if {@code queueDescription} is null.
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/QueueRuntimeInfo.java:34]: Change this to `queueProperties`.

File: sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusManagementClientTest.java [hunk 14]
```diff
@@ -266,14 +275,14 @@ void updateQueue() {
     @Test
     void updateQueueWithResponse() {
         // Arrange
-        final QueueDescription description = mock(QueueDescription.class);
-        final QueueDescription result = mock(QueueDescription.class);
+        final QueueProperties description = mock(QueueProperties.class);
```
Comment (srnagar) [inline sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusManagementClientTest.java:278]: Update the variable names to queueProperties.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionEntryContent.java (hunk 5 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescriptionEntryContent.java:60]: These are auto-generated from AutoRest. In another PR, I'm going to pull the actual QueueDescription class into implementation and have a separate QueueProperties class.

File: sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/implementation/EntityHelperTest.java [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+            .setDefaultMessageTimeToLive(Duration.ofSeconds(50))
+            .setDeadLetteringOnMessageExpiration(true)
+            .setDuplicateDetectionHistoryTimeWindow(Duration.ofSeconds(13))
+            .setEnableBatchedOperations(false)
+            .setEnablePartitioning(true)
+            .setForwardTo("Forward-To-This-Queue")
+            .setForwardDeadLetteredMessagesTo("Dead-Lettered-Forward-To")
+            .setLockDuration(Duration.ofSeconds(120))
+            .setMaxDeliveryCount(15)
+            .setMaxSizeInMegabytes(2048)
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/implementation/EntityHelperTest.java:34]: If we set negative values to these numbers, We should validate these values.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java [hunk 16]
```diff
@@ -0,0 +1,497 @@
...
+    /**
+     * Set the duplicateDetectionHistoryTimeWindow property: ISO 8601 timeSpan structure that defines the duration of
+     * the duplicate detection history. The default value is 10 minutes.
+     *
+     * @param duplicateDetectionHistoryTimeWindow the duplicateDetectionHistoryTimeWindow value to set.
+     *
+     * @return the CreateQueueOptions object itself.
+     */
+    public CreateQueueOptions setDuplicateDetectionHistoryTimeWindow(Duration duplicateDetectionHistoryTimeWindow) {
+        this.duplicateDetectionHistoryTimeWindow = duplicateDetectionHistoryTimeWindow;
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java:246]: I see that generally, there is no Null check, is this by design ?

Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java:228]: Yes. I try to do as little client validation if the service can do it. No value is === "use the default service value". It's not an invalid value.

File: sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/implementation/EntityHelperTest.java (hunk 15 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/implementation/EntityHelperTest.java:35]: Our guidelines say to do as little validation as possible and offset it to the service. I'd expect the service to do this validation and throw.

**Pull request merged**